panther_analysis_tool/main.py

```diff
@@ -1055,21 +1044,32 @@ def classify_analysis(
                         analysis_id, invalid_table_names
                     )
             analysis_ids.append(analysis_id)
+
+            classified_analysis = ClassifiedAnalysis(
+                analysis_spec_filename, dir_name, analysis_spec
+            )
+
             # extra validation for simple detections based on json schema
             if is_simple_detection(analysis_spec):
                 jsonschema.validate(analysis_spec, SIMPLE_DETECTION_SCHEMA)
-                classified_specs[SIMPLE_DETECTION].append(
-                    ClassifiedAnalysis(analysis_spec_filename, dir_name, analysis_spec)
-                )
+                all_specs.simple_detections.append(classified_analysis)
             # add the validated analysis type to the classified specs
-            elif analysis_type in [POLICY, RULE, SCHEDULED_RULE]:
-                classified_specs[DETECTION].append(
-                    ClassifiedAnalysis(analysis_spec_filename, dir_name, analysis_spec)
-                )
-            else:
-                classified_specs[analysis_type].append(
-                    ClassifiedAnalysis(analysis_spec_filename, dir_name, analysis_spec)
-                )
+            elif analysis_type in [
+                AnalysisTypes.POLICY,
+                AnalysisTypes.RULE,
+                AnalysisTypes.SCHEDULED_RULE,
+            ]:
+                all_specs.detections.append(classified_analysis)
+            elif analysis_type == AnalysisTypes.DATA_MODEL:
+                all_specs.data_models.append(classified_analysis)
+            elif analysis_type == AnalysisTypes.GLOBAL:
+                all_specs.globals.append(classified_analysis)
+            elif analysis_type == AnalysisTypes.LOOKUP_TABLE:
+                all_specs.lookup_tables.append(classified_analysis)
+            elif analysis_type == AnalysisTypes.PACK:
+                all_specs.packs.append(classified_analysis)
+            elif analysis_type == AnalysisTypes.SCHEDULED_QUERY:
+                all_specs.scheduled_queries.append(classified_analysis)
```

Comment:
`[minor]` hmm this logic feels like it might be better suited being in that new container class you created

Reply:
Good point. I will move it